### PR TITLE
Bug-fix for remote execution from windows using posixpath

### PR DESCRIPTION
**Pull request by aeshub**
## What changes are proposed in this pull request?

Changed os.path operations in databricks.py to posixpath to fix remote execution problem on Windows described in #1926 

## How is this patch tested?

There have been problems with the testing so far as described in #1926. Thus, Travis will hopefully be able to do the testing!

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

### What component(s) does this PR affect?

- [ ] UI
- [ ] CLI
- [ ] API
- [ ] REST-API
- [ ] Examples
- [ ] Docs
- [ ] Tracking
- [x] Projects
- [ ] Artifacts
- [ ] Models
- [ ] Scoring
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [x] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/projects/databricks.py [hunk 1]
```diff
@@ -120,20 +120,20 @@ def _upload_project_to_dbfs(self, project_dir, experiment_id):
                             a directory containing an MLproject file).
         """
         temp_tarfile_dir = tempfile.mkdtemp()
-        temp_tar_filename = os.path.join(temp_tarfile_dir, "project.tar.gz")
+        temp_tar_filename = posixpath.join(temp_tarfile_dir, "project.tar.gz")
```
Comment (smurching) [inline mlflow/projects/databricks.py]:
```suggestion
        temp_tar_filename = os.path.join(temp_tarfile_dir, "project.tar.gz")
```

A little tricky, but this path is on the local filesystem so we should use the OS's path join logic here

File: mlflow/projects/databricks.py [hunk 2]
```diff
@@ -120,20 +120,20 @@ def _upload_project_to_dbfs(self, project_dir, experiment_id):
...
 
         def custom_filter(x):
-            return None if os.path.basename(x.name) == "mlruns" else x
+            return None if posixpath.basename(x.name) == "mlruns" else x
```
Comment (smurching) [inline mlflow/projects/databricks.py]:
```suggestion
            return None if os.path.basename(x.name) == "mlruns" else x
```

As above, we should use the OS's path logic for local files

**Review by smurching — COMMENTED** [review]
Two tiny comments (left suggestions that when committed should address my comments) but otherwise LGTM! Were you able to try running a project with this modified logic?

**Comment by aestene** [comment]
Good point! Remote execution is working well on Databricks now. I have run it with the latest changes back to os.path and there is no problem. I've only been able to check on Windows, though I see no reason why this would break anything on Linux :)

**Review by smurching — APPROVED** [review]
LGTM thanks for the updates @aestene! And thanks for doing the QA on Windows :)

**Pull request merged**